aiida/engine/processes/workchains/workchain.py

```diff
@@ -167,6 +167,11 @@ def resolve_awaitable(self, awaitable: Awaitable, value: Any) -> None:
 
         awaitable.resolved = True
 
+        if self.has_terminated:
```

Comment:
to make it even more explicit (only update, if process not terminated), you could say

```
if not self.has_terminated:
   self._update_process_status()
```

Reply:
done